### ISSUE TITLE
build error: modpost: GPL symbol 'alternatives_patched'

### ISSUE THREAD

**Issue by ste7677** [zfsonlinux/zfs#9041]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	| gentoo
Distribution Version	| 17.1 / GCC 9
Linux Kernel	|  5.2.1
Architecture	|  x86_64
ZFS Version	|  master @ df834a7 
SPL Version	| 

### Describe the problem you're observing
zfs-kmod build error:
```
make -f ./scripts/Makefile.modpost
  find /var/tmp/portage/sys-fs/zfs-kmod-9999/work/zfs-kmod-9999/module/.tmp_versions -name '*.mod' | xargs -r grep -h '\.ko$' | sort -u | sed 's/\.ko$/.o/' | scripts/mod/modpost   -i ./Module.symvers -I /var/tmp/portage/sys-fs/zfs-kmod-9999/work/zfs-kmod-9999/module/Module.symvers  -o /var/tmp/portage/sys-fs/zfs-kmod-9999/work/zfs-kmod-9999/module/Module.symvers    -s -T -
FATAL: modpost: GPL-incompatible module zfs.ko uses GPL-only symbol 'alternatives_patched'
make[4]: *** [scripts/Makefile.modpost:91: __modpost] Error 1
make[3]: *** [Makefile:1599: modules] Error 2
make[3]: Leaving directory '/usr/src/linux-5.2.1-gentoo'
make[2]: *** [Makefile:30: modules] Error 2
make[2]: Leaving directory '/var/tmp/portage/sys-fs/zfs-kmod-9999/work/zfs-kmod-9999/module'
make[1]: *** [Makefile:840: all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/sys-fs/zfs-kmod-9999/work/zfs-kmod-9999'
make: *** [Makefile:709: all] Error 2
 * ERROR: sys-fs/zfs-kmod-9999::gentoo failed (compile phase):
 *   emake failed
```
building zfs-kmod release 0.8.1 against kernel 5.2.1 works (only build tested, not the functionality)

see kernel .config and complete build log attached:

[kernel.config.txt](https://github.com/zfsonlinux/zfs/files/3396855/kernel.config.txt)
[build.log](https://github.com/zfsonlinux/zfs/files/3396856/build.log)

**Comment by drJeckyll** [comment]
It fails to build with gcc 8.2.0-r4 too with 5.2.0 kernel. Couple of days ago (actually on 11-th July) with same 5.2.0 kernel it manage to build and it works.

**Comment by satmandu** [comment]
Probably related to this patch: https://github.com/zfsonlinux/zfs/commit/e5db31349484e5e859c7a942eb15b98d68ce5b4d

**Comment by satmandu** [comment]
Here's where that shows up in the kernel: https://github.com/torvalds/linux/blob/master/arch/x86/include/asm/alternative.h

**Comment by satmandu** [comment]
Reverting https://github.com/zfsonlinux/zfs/commit/e5db31349484e5e859c7a942eb15b98d68ce5b4d allows the compile to succeed.

**Comment by behlendorf** [comment]
Are there additional changes in this area applied to the Gentoo kernel?  This isn't a symbol we call directly, or even indirectly as far as I can tell.  We haven't observed this build failure against the Fedora 5.2 kernel, or even a vanilla 5.3.0-0.rc0.

**Comment by satmandu** [comment]
Saw this error earlier today when building zfs-dkms compiled from master with the 5.2 Ubuntu eoan kernel in testing.

**Comment by behlendorf** [comment]
I believe I see, it's the result of the kernel being compiled with `CONFIG_X86_DEBUG_FPU`.  I'll see what can be done.

**Comment by ste7677** [comment]
Confirmed. `CONFIG_X86_DEBUG_FPU` has already been enabled on an earlier kernel 5.1.3.-gentoo. Earlier zfs-kmod compiled fine.

Disabling `CONFIG_X86_DEBUG_FPU` temporary fixes the issue. zfs-kmod compiles fine.

**Comment by behlendorf** [comment]
@ste7677 thanks for verifying this.  Resolved by  095b541.

**Comment by ste7677** [comment]
Compiles fine from master now with `CONFIG_X86_DEBUG_FPU` enabled. 

@behlendorf  Thank you!
